components/api/src/main/java/com/hotels/styx/api/support/UUIDGenerator.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.api.support;
+
+import java.util.UUID;
+
+import com.fasterxml.uuid.EthernetAddress;
+import com.fasterxml.uuid.Generators;
+import com.fasterxml.uuid.NoArgGenerator;
+import com.fasterxml.uuid.UUIDComparator;
+
+/**
+ * Utility class to make a {@link java.util.UUID}.
+ */
+public class UUIDGenerator {
+    private static final NoArgGenerator TIME_BASED_GENERATOR = Generators.timeBasedGenerator(EthernetAddress.fromInterface());
+
+    public static UUID makeUUID() {
+        return TIME_BASED_GENERATOR.generate();
+    }
+
+    public static boolean uuidsEqual(UUID u1, UUID u2) {
```

Comment:
Relying on developers to use this static method instead of `Object.equals` is likely to result in human error during future development, particular when used indirectly, e.g. as the key in a `HashMap` or as a member of a `HashSet`.
Instead, I would suggest creating a new `UUID` class that wraps an instance of `java.util.UUID`, but uses your `equals` implementation.